.changeset/tall-waves-impress.md

```diff
@@ -0,0 +1,58 @@
+---
+'astro': minor
+---
+
+Change Action cookie redirects to an opt-in feature users can implement using middleware. This means Actions submitted from an HTML form action will no longer redirect to the destination as a GET request. Instead, the page will be rendered as a POST result.
+
+This is meant to address the 4 KB size limit users have encountered when calling actions from an HTML form action. It can be difficult to predict the size of an action result for large validation errors or longer return values, like AI chatbot results.
+
+So, we've introduced a new `getActionContext()` utility to let you decide how action results are handled from middleware. You may choose to implement the existing cookie redirect from Astro v4, or implement forwarding with your own session storage.
+
+## Migration
+
+Cookie redirects are no longer handled by default. You can implement the behavior from Astro v4 as a middleware using `getActionContext()` like so:
```

Comment:
```suggestion
Cookie redirects are no longer handled by default. Update an existing Astro v4 action as a middleware using `getActionContext()`:
```
Maybe slight preference for framing this as "updating existing actions" since that's what people will be doing, and that feels like reaching them where they're at? If people haven't yet got an action, they'll just read the docs and see all the current advice there?

Reply:
Eh, I'm not sure that phrasing makes sense. I'll pull from the entry I wrote for the 5.0 Beta docs, which I think describes this more clearly than the changeset